tools/PI/DevHome.PI/Helpers/WatsonHelper.cs

```diff
@@ -86,49 +81,28 @@ public void EventLogEventRead(object? obj, EventRecordWrittenEventArgs eventArg)
                     var moduleName = eventRecord.Properties[3].Value.ToString() ?? string.Empty;
                     var executable = eventRecord.Properties[0].Value.ToString() ?? string.Empty;
                     var eventGuid = eventRecord.Properties[12].Value.ToString() ?? string.Empty;
-                    var report = new WatsonReport(timeGenerated, moduleName, executable, eventGuid);
-                    watsonOutput?.Add(report);
 
-                    WinLogsEntry entry = new(timeGenerated, WinLogCategory.Error, eventRecord.FormatDescription(), WinLogsHelper.WatsonName);
-                    winLogsPageOutput?.Add(entry);
+                    var watsonViewModel = Application.Current.GetService<WatsonPageViewModel>();

```

Comment:
>var watsonViewModel  [](http://example.com/codeflow?start=20&length=20)

I don't think this is the right model to use.... our Watson Helper shouldn't know anything about the view model.... it should expose Watson reports that the various view model can consume/filter.

I have a branch where I have this laid out.

[https://github.com/microsoft/devhome/tree/user/timkurNewWatson](https://github.com/microsoft/devhome/tree/user/timkurNewWatson)